File: spec/php/UserTypeTest.php
<?php
namespace Kaitai\Struct\Tests;

class UserTypeTest extends TestCase {
    public function testUserType() {
        $r = UserType::fromFile(self::SRC_DIR_PATH . "/repeat_until_s4.bin");

        $this->assertEquals(0x42, $r->one->width);
        $this->assertEquals(0x1337, $r->one->height);
    }
}

File: spec/go/user_type_test.go
package spec

import (
	"os"
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"

	. "test_formats"
)

func TestUserType(t *testing.T) {
	f, err := os.Open("../../src/repeat_until_s4.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r UserType
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, 0x42, r.One.Width)
	assert.EqualValues(t, 0x1337, r.One.Height)
}
